pkg/tsdb/loki/frame.go

```diff
@@ -81,53 +83,49 @@ func adjustMetricFrame(frame *data.Frame, query *lokiQuery) error {
 func adjustLogsFrame(frame *data.Frame, query *lokiQuery) error {
 	// we check if the fields are of correct type and length
 	fields := frame.Fields
-	if len(fields) != 3 {
+	if len(fields) != 4 {
 		return fmt.Errorf("invalid fields in logs frame")
 	}
 
 	labelsField := fields[0]
 	timeField := fields[1]
 	lineField := fields[2]
+	stringTimeField := fields[3]
 
-	if (timeField.Type() != data.FieldTypeTime) || (lineField.Type() != data.FieldTypeString) || (labelsField.Type() != data.FieldTypeJSON) {
+	if (timeField.Type() != data.FieldTypeTime) || (lineField.Type() != data.FieldTypeString) || (labelsField.Type() != data.FieldTypeJSON) || (stringTimeField.Type() != data.FieldTypeString) {
 		return fmt.Errorf("invalid fields in logs frame")
 	}
 
-	if (timeField.Len() != lineField.Len()) || (timeField.Len() != labelsField.Len()) {
+	if (timeField.Len() != lineField.Len()) || (timeField.Len() != labelsField.Len()) || (timeField.Len() != stringTimeField.Len()) {
```

Comment:
i think you can use [frame.RowLen()](https://github.com/grafana/grafana-plugin-sdk-go/blob/ab60be44b4c840fc51a56b02a6d3328327316218/data/frame.go#L262-L264) to check that the fields lengths match

Reply:
nice, done!